Fix up LDFLAGS for Precise and earlier (patch by Hans-Christoph Steiner,
updated to pd-l2ork by Albert Graef).
Description: fix ./configure dying thinking we're cross-compiling
 When building using debuild on Debian/squeeze or Ubuntu < quantal,
 ./configure somehow gets some LDFLAGS for shared libraries and uses them in
 the gcc tests.  The executable then fails to run since its a shared library,
 and ./configure thinks we're cross-compiling.  This patch forces no LDFLAGS.

diff --git a/externals/Makefile b/externals/Makefile
index 7b64f45..6b7c49b 100644
--- a/externals/Makefile
+++ b/externals/Makefile
@@ -611,6 +611,7 @@ $(gem_src)/Gem.pd_linux: $(gem_src)/configure
 	test -s $(gem_src)/Gem.pd_linux || \
 		cd $(gem_src) && ./configure \
 			CXXFLAGS="-DHAVE_S_STUFF_H" \
+			LDFLAGS= \
 			--prefix=$(prefix) \
 			--with-video=plugins \
 			--with-film=plugins \
@@ -649,7 +650,7 @@ $(externals_src)/gem2pdp/configure: $(externals_src)/gem2pdp/configure.ac
 	cd $(externals_src)/gem2pdp && aclocal && autoconf
 
 $(externals_src)/gem2pdp/Makefile: $(externals_src)/gem2pdp/Makefile.in
-	cd $(externals_src)/gem2pdp && ./configure --with-pddir=$(pd_src) \
+	cd $(externals_src)/gem2pdp && ./configure LDFLAGS= --with-pddir=$(pd_src) \
 		--with-gemdir=$(gem_src)  --with-pdpdir=$(externals_src)/pdp
 
 gem2pdp: $(externals_src)/gem2pdp/configure $(externals_src)/gem2pdp/Makefile
@@ -885,7 +886,7 @@ $(IEM16_DIR)/configure: $(IEM16_DIR)/configure.ac $(IEM16_DIR)/aclocal.m4
 
 $(IEM16_DIR)/Make.config: $(IEM16_DIR)/Make.config.in \
 $(IEM16_DIR)/configure
-	cd $(IEM16_DIR) && ./configure --disable-library \
+	cd $(IEM16_DIR) && ./configure LDFLAGS= --disable-library \
 		--with-pd=$(pd_src)
 
 iem16: $(IEM16_DIR)/Make.config
@@ -1269,11 +1270,11 @@ $(externals_src)/OSCx/configure: $(externals_src)/OSCx/configure.ac
 	cd $(externals_src)/OSCx && autoconf
 
 $(externals_src)/OSCx/Makefile: $(externals_src)/OSCx/Makefile.in
-	cd $(externals_src)/OSCx && ./configure
+	cd $(externals_src)/OSCx && ./configure LDFLAGS=
 $(externals_src)/OSCx/libOSC/Makefile: $(externals_src)/OSCx/libOSC/Makefile.in
-	cd $(externals_src)/OSCx && ./configure
+	cd $(externals_src)/OSCx && ./configure LDFLAGS=
 $(externals_src)/OSCx/src/Makefile: $(externals_src)/OSCx/src/Makefile.in
-	cd $(externals_src)/OSCx && ./configure
+	cd $(externals_src)/OSCx && ./configure LDFLAGS=
 
 $(externals_src)/OSCx/src/OSCroute.$(EXTENSION):  $(externals_src)/OSCx/configure \
 $(externals_src)/OSCx/Makefile
@@ -1356,7 +1357,7 @@ $(externals_src)/pdp/configure: $(externals_src)/pdp/configure.ac
 	cd $(externals_src)/pdp && autoconf
 
 $(externals_src)/pdp/Makefile.config: $(externals_src)/pdp/configure $(externals_src)/pdp/Makefile.config.in
-	cd $(externals_src)/pdp && ./configure PD_CPPFLAGS="-I$(pd_src)/src" \
+	cd $(externals_src)/pdp && ./configure LDFLAGS= PD_CPPFLAGS="-I$(pd_src)/src" \
 		$(PDP_OPTIONS)
 
 $(externals_src)/pdp/pdp.$(EXTENSION): $(externals_src)/pdp/Makefile.config 
@@ -1770,6 +1771,7 @@ $(zexy_src)/configure: $(zexy_src)/configure.ac
 
 $(zexy_src)/Makefile: $(zexy_src)/configure $(zexy_src)/Makefile.am $(zexy_src)/src/Makefile.am
 	cd $(zexy_src) && ./configure --disable-library \
+		LDFLAGS= \
 		--with-extension=$(EXTENSION) \
 		--prefix=$(prefix) \
 		--libdir=$(objectsdir) \
diff --git a/externals/iem16/src/Makefile b/externals/iem16/src/Makefile
index 25e5bbe..5be876d 100644
--- a/externals/iem16/src/Makefile
+++ b/externals/iem16/src/Makefile
@@ -22,7 +22,7 @@ aclocal.m4: acinclude.m4
 	aclocal
 
 Make.config: Make.config.in configure
-	./configure $(CONFIGUREFLAGS)
+	./configure $(CONFIGUREFLAGS) LDFLAGS=
 
 -include $(SOURCES:.c=.d)
 
diff --git a/externals/moocow/extended/Makefile b/externals/moocow/extended/Makefile
index 2726d1a..9da4944 100644
--- a/externals/moocow/extended/Makefile
+++ b/externals/moocow/extended/Makefile
@@ -36,7 +36,7 @@ MOOCOW_BUILD_VERSION ?=2009-04-27.002
 #	CFLAGS="$(shell echo $(CFLAGS))" ##-- works
 CONFIGURE_ARGS=\
 	CFLAGS="$(CFLAGS) -I/sw/include" \
-	LDFLAGS="$(LDFLAGS) -L$(pd_src)/src -L$(pd_src)/bin -L$(pd_src)/obj -L/sw/lib" \
+	LDFLAGS="-L$(pd_src)/src -L$(pd_src)/bin -L$(pd_src)/obj -L/sw/lib" \
 	--with-pd-include="$(pd_src)/src" \
 	--with-pd-dir="$(MOOCOW_BUILD)" \
 	--disable-dependency-tracking
